packages/block-editor/src/components/block-navigation/list.js

```diff
@@ -40,30 +46,16 @@ export default function BlockNavigationList( {
 		/* eslint-disable jsx-a11y/no-redundant-roles */
 		<ul className="block-editor-block-navigation__list" role="list">
 			{ map( omitBy( blocks, isNil ), ( block ) => {
-				const blockType = getBlockType( block.name );
 				const isSelected = block.clientId === selectedBlockClientId;
-
 				return (
 					<li key={ block.clientId }>
-						<div className="block-editor-block-navigation__item">
-							<Button
-								className={ classnames(
-									'block-editor-block-navigation__item-button',
-									{
-										'is-selected': isSelected,
-									}
-								) }
-								onClick={ () => selectBlock( block.clientId ) }
-							>
-								<BlockIcon icon={ blockType.icon } showColors />
-								{ getBlockLabel( blockType, block.attributes ) }
-								{ isSelected && (
-									<VisuallyHidden as="span">
-										{ __( '(selected block)' ) }
-									</VisuallyHidden>
-								) }
-							</Button>
-						</div>
+						<ListItemWrapper
```

Comment:
Naming wise, it stood out to me that this is `ListItemWrapper` and yet it doesn't wrap the list item (`<li>`), though I guess it wraps `BlockNavigationListItem`.

I personally feel like making the slot implementation internal to `BlockNavigationListItem` and then making that component represent the whole `<li>` might provide a better separation:
```jsx
<BlockNavigationListItem
    block={ block }
    isSelected={ block.clientId === selectedBlockClientId }
    // ... other props
 />
        { showNestedBlocks &&
		!! block.innerBlocks &&
		!! block.innerBlocks.length && (
			<BlockNavigationList
				blocks={ block.innerBlocks }
				selectedBlockClientId={
					selectedBlockClientId
				}
				selectBlock={ selectBlock }
				parentBlockClientId={ block.clientId }
				showAppender={ showAppender }
				showNestedBlocks
			/>
		) }
</BlockNavigationListItem>
```

Using a weird notation that I just made up, the hierarchy would be something like:

- `BlockNavigationList` renders many
  - `BlockNavigationListItem` renders
    - `useBlockNavigationSlots` ? `BlockNavigationItemSlot` : `BlockNavigationItem`
    - `BlockNavigationList` (as `children`)